crates/iota-node/src/lib.rs

```diff
@@ -833,6 +843,34 @@ impl IotaNode {
         let iota_node_metrics =
             Arc::new(IotaNodeMetrics::new(&registry_service.default_registry()));
 
+        // --- Create shared checkpoint broadcast channel and buffer for gRPC and
+        // checkpoint logic ---
+        let (grpc_checkpoint_summary_tx, grpc_checkpoint_data_tx) = if let Some(grpc_api_address) =
+            config.grpc_api_address
+        {
+            let (summary_tx, _) = tokio::sync::broadcast::channel(CHECKPOINT_BROADCAST_BUFFER_SIZE);
+            let (data_tx, _) = tokio::sync::broadcast::channel(CHECKPOINT_BROADCAST_BUFFER_SIZE);
+            let rocks = RocksDbStore::new(
```

Comment:
Wait, why do we create a new RocksDbStore here? Will that even work if we have two at the same time?
Why don't we reuse the `state_sync_store`?